src/odb/src/3dblox/dbvWriter.cpp

```diff
@@ -0,0 +1,359 @@
+             << YAML::EndSeq;
+      YAML::Node layer_id = cad_layer["108;103"];
+      layer_id["type"] = "scribe_line";
+      layer_id["scribe_line_remaining_width"] = YAML::Load(sc_out.c_str());
+    } else {
+      YAML::Node layer_id = cad_layer["108;101"];
+      layer_id["type"] = "design_area";
+    }
+  }
+  // External files
+  YAML::Node external_node = chiplet_node["external"];
+  writeExternal(external_node, chiplet, db);
+
+  // Regions
+  YAML::Node regions_node = chiplet_node["regions"];
+  writeRegions(regions_node, chiplet, db);
+}
+
+void DbvWriter::writeRegions(YAML::Node& regions_node,
+                             odb::dbChip* chiplet,
+                             odb::dbDatabase* db)
+{
+  for (auto region : chiplet->getChipRegions()) {
+    YAML::Node region_node = regions_node[region->getName()];
+    writeRegion(region_node, region, db);
+  }
+}
+
+static const char* chip_region_side_to_string(odb::dbChipRegion::Side side)
+{
+  switch (side) {
+    case odb::dbChipRegion::Side::FRONT:
+      return "front";
+    case odb::dbChipRegion::Side::BACK:
+      return "back";
+    case odb::dbChipRegion::Side::INTERNAL:
+      return "internal";
+    case odb::dbChipRegion::Side::INTERNAL_EXT:
+      return "internal_ext";
+  }
+  return "front";
+}
+
+void DbvWriter::writeRegion(YAML::Node& region_node,
+                            odb::dbChipRegion* region,
+                            odb::dbDatabase* db)
+{
+  region_node["side"] = chip_region_side_to_string(region->getSide());
+  if (auto layer = region->getLayer(); layer != nullptr) {
+    region_node["layer"] = layer->getName();
+  }
+  YAML::Node coords_node = region_node["coords"];
+  writeCoordinates(coords_node, region->getBox(), db);
+}
+
+void DbvWriter::writeExternal(YAML::Node& external_node,
+                              odb::dbChip* chiplet,
+                              odb::dbDatabase* db)
+{
+  BaseWriter::writeLef(external_node, db, chiplet);
+  BaseWriter::writeDef(external_node, db, chiplet);
+}
+
+void DbvWriter::writeCoordinates(YAML::Node& coords_node,
+                                 const odb::Rect& rect,
+                                 odb::dbDatabase* db)
+{
+  const double u = db->getDbuPerMicron();
+  YAML::Emitter c0;
+  c0 << YAML::Flow << YAML::BeginSeq << rect.xMin() / u << rect.yMin() / u
+     << YAML::EndSeq;
+  YAML::Emitter c1;
+  c1 << YAML::Flow << YAML::BeginSeq << rect.xMax() / u << rect.yMin() / u
+     << YAML::EndSeq;
+  YAML::Emitter c2;
+  c2 << YAML::Flow << YAML::BeginSeq << rect.xMax() / u << rect.yMax() / u
+     << YAML::EndSeq;
+  YAML::Emitter c3;
+  c3 << YAML::Flow << YAML::BeginSeq << rect.xMin() / u << rect.yMax() / u
+     << YAML::EndSeq;
+  coords_node.push_back(YAML::Load(c0.c_str()));
+  coords_node.push_back(YAML::Load(c1.c_str()));
+  coords_node.push_back(YAML::Load(c2.c_str()));
+  coords_node.push_back(YAML::Load(c3.c_str()));
```

Comment:
why write to string then read string? I think there is a way to write a yaml sequence directly